sim/paladin/holy_power.go

```diff
@@ -53,6 +64,11 @@ func (pb *HolyPowerBar) SpendHolyPower(sim *core.Simulation, metrics *core.Resou
 		return
 	}
 
+	if pb.paladin.DivinePurposeAura.IsActive() {
+		pb.paladin.Log(sim, "Consumed Divine Purpose")
+		return
+	}
```

Comment:
Do you not need to deactivate/consume the DP aura?

Reply:
I did it the way it was done for Art of War (https://github.com/wowsims/cata/blob/21c8cc30a7b58134652dbf68eb8112058b2984ee/sim/paladin/retribution_talents.go#L193) which might not be correct?
Seemed a bit disconnected when I saw it but I just went with how it was done there. Would feel cleaner to deactivate both DP and AoW in their respective places I guess :)

Reply:
Ok the way it deactivates is the correct way (keeps the DP code contained in one place). Now what I dont like is this check for DP in the holy_power code. I wonder if this can be avoided somehow. I guess it can become quite more complicated to track HP and refund it after the spend when deactivating the HP but i'm not sure its worth it. Maybe how you have it is better even if the separation of concerns is a bit worse

Reply:
The cleaner way would be if HP was handled like a resource and DP could make it cost 0 with a spell mod.
The problem with this (other than having two power bars) is that just like with CP for rogues/druids, TV and Inq consume all available HP every time so it would make no sense to configure any "HolyPowerCost" on an ability.